File: specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/examples-js/RedisCacheImport.js
const { RedisManagementClient } = require("@azure/arm-rediscache");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Import data into Redis cache.
 *
 * @summary Import data into Redis cache.
 * x-ms-original-file: specification/redis/resource-manager/Microsoft.Cache/stable/2022-06-01/examples/RedisCacheImport.json
 */
async function redisCacheImport() {
  const subscriptionId = "subid";
  const resourceGroupName = "rg1";
  const name = "cache1";
  const parameters = {
    format: "RDB",
    files: ["http://fileuris.contoso.com/pathtofile1"],
  };
  const credential = new DefaultAzureCredential();
  const client = new RedisManagementClient(credential, subscriptionId);
  const result = await client.redis.beginImportDataAndWait(resourceGroupName, name, parameters);
  console.log(result);
}

redisCacheImport().catch(console.error);

File: specification/redis/resource-manager/Microsoft.Cache/stable/2024-03-01/examples-java/RedisCacheImport.java

import com.azure.resourcemanager.redis.models.ImportRdbParameters;
import java.util.Arrays;

/**
 * Samples for Redis ImportData.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/redis/resource-manager/Microsoft.Cache/stable/2024-03-01/examples/RedisCacheImport.json
     */
    /**
     * Sample code: RedisCacheImport.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void redisCacheImport(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.redisCaches().manager().serviceClient().getRedis().importData("rg1", "cache1",
            new ImportRdbParameters().withFormat("RDB")
                .withFiles(Arrays.asList("http://fileuris.contoso.com/pathtofile1"))
                .withStorageSubscriptionId("storageSubId"),
            com.azure.core.util.Context.NONE);
    }
}
